### ISSUE TITLE
Header & footer widgets can't be synced to sn-scriptsync

### ISSUE THREAD

**Issue by Blenderpics** [arnoudkooi/ServiceNow-Utils#23]
Hey!
just noticed a weird quirk:
Wenn you try to hit the sync button in the widget editor while a service portal header/footer is opened, the widget does not get synced.

The following error gets thrown in inject.js:820  :
`Uncaught TypeError: Cannot set property 'choices' of undefined  
    at postRequestToScriptSync (inject.js:820)  
    at HTMLButtonElement.onclick (sp_config? 
 id=widget_editor&sys_id=3121edb44f4363009bc6df601310c7ec&spa=1:1)`

Im using the latest chrome store release of the addon + ServiceNow Kingston

**Comment by arnoudkooi** [comment]
Could not replicate it, are you editing a clone of a ootb widget? What is the value of the Data Table field?

**Comment by Blenderpics** [comment]
The value of the data table field is "Instance [sp_instance]".
I tested multiple headers (cloned stock widgets and widgets which where build from scratch). All throw the same error when i try to sync them.  
I forgot to mention that i work in a scoped application.  
I am not a instance admin either. I just have permissions for one scoped application.  
Not sure if this changes anything.

**Comment by arnoudkooi** [comment]
Maybe try after commenting line 820 in inject.js in dev tools

**Comment by Blenderpics** [comment]
I'll give it a shot tomorrow.
Thanks for the support so far!

**Comment by arnoudkooi** [comment]
Please check if it works now, after updating to version 3.2.0.1 I made added a check if data_tables property is present (linenumber is changed)

**Comment by Blenderpics** [comment]
Because of the hasOwnProperty check, there is no error thrown anymore.  
However, the widget still does not get synced.
`client.send(...)` seems to be successfull, since `client.onreadystatechange` is invoked with `client.readyState == 2` and `client.status == 200`.  
So it now seems like sn-scriptsync failes somewhere?  
I could not figure out how to debug vs code addons, so my debugging session ended here.  
One interesting thing i noticed is, that the properties `angularData.f._fields` look quite different on header/footer widgets compared to normal widgets.  
![widget_bug](https://user-images.githubusercontent.com/1348725/51438326-0e839300-1cab-11e9-8635-c93298d874c2.png)

I start to think this might be an instance specific problem.

**Comment by arnoudkooi** [comment]
Can you share the XML of a faulty widget?

**Comment by Blenderpics** [comment]
Sure
```<?xml version="1.0" encoding="UTF-8"?>
<record_update table="sp_header_footer">
	<sp_header_footer action="INSERT_OR_UPDATE">
		<category>custom</category>
		<client_script>
			<![CDATA[function() {
  /* widget controller */
  var c = this;
}]]>
		</client_script>
		<controller_as>c</controller_as>
		<css>.thisClassIsNotUsed {&#13;
  &#13;
}</css>
		<data_table>sp_instance</data_table>
		<demo_data/>
		<description/>
		<docs/>
		<field_list/>
		<has_preview>false</has_preview>
		<id>mp_test</id>
		<internal>false</internal>
		<link/>
		<name>MP Test</name>
		<option_schema/>
		<public>false</public>
		<roles/>
		<script>
			<![CDATA[(function() {
  /* populate the 'data' object */
  /* e.g., data.table = $sp.getValue('table'); */

})();]]>
		</script>
		<servicenow>false</servicenow>
		<sys_class_name>sp_header_footer</sys_class_name>
		<sys_created_by>[REMOVED]</sys_created_by>
		<sys_created_on>2019-01-20 15:35:14</sys_created_on>
		<sys_customer_update>false</sys_customer_update>
		<sys_id>62914d3e4f8b6b005b02bc511310c7ed</sys_id>
		<sys_mod_count>0</sys_mod_count>
		<sys_name>MP Test</sys_name>
		<sys_package display_value="[REMOVED]" source="[REMOVED]">219282484f916f009bc6df601310c7dd</sys_package>
		<sys_policy/>
		<sys_replace_on_upgrade>false</sys_replace_on_upgrade>
		<sys_scope display_value="[REMOVED]">219282484f916f009bc6df601310c7dd</sys_scope>
		<sys_update_name>sp_header_footer_62914d3e4f8b6b005b02bc511310c7ed</sys_update_name>
		<sys_updated_by>[REMOVED]</sys_updated_by>
		<sys_updated_on>2019-01-20 15:35:14</sys_updated_on>
		<template>
			<![CDATA[<div>
<!-- your widget template -->
</div>]]>
		</template>
	</sp_header_footer>
</record_update>```

**Comment by arnoudkooi** [comment]
I was not aware this was in an extended table sp_header_footer. I just published ScriptSync 1.1.3 Please give that a try (note that the widgets will be stored in the sp_widget folder. Not in sp_header_footer)

**Comment by Blenderpics** [comment]
Works great! Thanks!
